VOKMockUrlProtocol.m

```diff
@@ -111,6 +118,28 @@ - (NSArray *)resourceNames
         [resourceName appendFormat:queryFormat, self.request.URL.query];
     }
     
+    if (headersToEncode) {
+        NSDictionary *headerDict = self.request.allHTTPHeaderFields;
+        NSMutableArray <NSString *>*allIncludedHeaders = [NSMutableArray array];
+        for (NSString *headerName in headersToEncode) {
+            NSString *headerValue = headerDict[headerName];
+            if (headerValue) {
+                NSString *headerString = [NSString stringWithFormat:@"%@=%@", headerName, headerValue];
+                [allIncludedHeaders addObject:headerString];
+            }
+        }
+        
+        NSString *fullHeaderString = [allIncludedHeaders componentsJoinedByString:@"&"];
+        if (fullHeaderString.length > 0) {
+            NSString *encodedHeaderString = [fullHeaderString stringByAddingPercentEscapesUsingEncoding:NSUTF8StringEncoding];
+            NSData *fullHeaderData = [encodedHeaderString dataUsingEncoding:NSUTF8StringEncoding];
+            NSString *hashedHeaderString = [self sha256HexOfData:fullHeaderData];
+        
+            [resourceNames addObject:[[resourceName stringByAppendingFormat:AppendSeparatorFormat, hashedHeaderString] mutableCopy]];
+            [resourceName appendFormat:AppendSeparatorFormat, encodedHeaderString];
```

Comment:
There may already be multiple resource names at this point, so I think you have to add your potential two additions to each existing resource name—take a look below, around lines 180–188.